zdm/survey.py

```diff
@@ -88,7 +84,8 @@ def get_efficiency_from_wlist(self,DMlist,wlist,plist,
         """
         efficiencies=np.zeros([wlist.size,DMlist.size])
         if addGalacticDM:
-            toAdd = self.DMhalo + np.mean(self.DMGs)
+            # the following is safe against surveys with zero FRBs
```

Comment:
How many of these changes need to propogate to the new `Survey` class?

Reply:
I've now fully debugged the new survey class to be robust against surveys with 0 and >0 FRBs. The methods are quite different in the new survey class.